### PR TITLE
Add ripple effect on Threads list cardview

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #103

### CODE REVIEW

File: app/src/main/res/layout/item_thread.xml [hunk 1]
```diff
@@ -15,128 +15,134 @@
   ~ You should have received a copy of the GNU General Public License
   ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
   -->
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<com.google.android.material.card.MaterialCardView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (sirambd) [inline app/src/main/res/layout/item_thread.xml:18]: This layout contains a lot of layouts, can't we simplify everything?

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadListAdapter.kt [hunk 2]
```diff
@@ -97,10 +97,9 @@ class ThreadListAdapter : RecyclerView.Adapter<ViewHolder>() { // TODO: Use Load
 
     private fun ItemThreadBinding.displayThread(position: Int) {
         val thread = itemsList[position] as Thread
-        val context = itemThread.context
 
         expeditor.text = thread.from[0].name.ifEmpty { thread.from[0].email }
-        mailSubject.text = thread.subject.displayedSubject(context)
+        mailSubject.text = thread.subject.displayedSubject(root.context)
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadListAdapter.kt:102]: use binding receiver context extension for the context

**Review by sirambd — CHANGES_REQUESTED** [review]
Waiting for request changes

File: app/src/main/res/layout/item_thread.xml (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/res/layout/item_thread.xml:18]: Will be handled by this issue: https://github.com/Infomaniak/android-mail/issues/142

**Pull request merged**